src/ogd/games/BLOOM/features/FailCount.py

```diff
@@ -0,0 +1,42 @@
+# import necessary libraries
+from typing import Any, Dict, List, Optional
+from ogd.core.generators.Generator import GeneratorParameters
+from ogd.core.generators.extractors.Feature import Feature
+from ogd.core.schemas.Event import Event
+from ogd.core.schemas.ExtractionMode import ExtractionMode
+from ogd.core.schemas.FeatureData import FeatureData
+
+class FailCountFeature(Feature):
+    def __init__(self, params: GeneratorParameters):
+        super().__init__(params=params)
+        self.fail_count: Dict[str, int] = {
+            "CityFailed": 0,
+            "TooManyBlooms": 0,
+            "OutOfMoney": 0
+        }
+
+    # *** IMPLEMENT ABSTRACT FUNCTIONS ***
+    @classmethod
+    def _eventFilter(cls, mode: ExtractionMode) -> List[str]:
+        return ["lose_game"]
+
+    @classmethod
+    def _featureFilter(cls, mode: ExtractionMode) -> List[str]:
+        return []
+
+    def _updateFromEvent(self, event: Event) -> None:
+        lose_condition = event.EventData.get("lose_condition", "")
```

Comment:
Use `None` instead of `""` as the default value of the `.get(...)` function call, otherwise the next line `if lose_condition:` will always evaluate to `True`.

Reply:
Nevermind, I misremembered Python's casting rules. This works out fine.